src/ecdh.rs

```diff
@@ -44,6 +47,31 @@ impl SharedSecret {
         }
     }
 
+    /// Creates a new shared secret from a pubkey and secret key with applied custom hash function
+    pub fn new_with_hash<F>(point: &PublicKey, scalar: &SecretKey, hash: &mut F) -> SharedSecret
+        where F: FnMut(&mut [u8], &[u8], &[u8]) -> i32
+    { 
+        extern "C" fn hash_callback<F>(output: *mut c_uchar, x: *const c_uchar, y: *const c_uchar, data: *const c_void) -> c_int 
+            where F: FnMut(&mut [u8], &[u8], &[u8]) -> i32
+        {
+            let callback: &mut F = unsafe { &mut *(data as *mut F) };
+            unsafe { (*callback)(from_raw_parts_mut(output, 32), from_raw_parts(x, 32), from_raw_parts(y, 32)) }
+        }
+        unsafe {
+            let mut ss = ffi::SharedSecret::new();
```

Comment:
Please minimize the code inside the unsafe block to the minimum possible needed (i.e. the ffi call *only*)